### PR TITLE
Inventory and Container Tweaks

### PR DESCRIPTION
**Pull request by qreeves**
* More efficient update routine
* Abstracted repeated getNumSlots
* Improved layout of inventory and containers.

### CODE REVIEW

File: engine/src/main/java/org/terasology/rendering/nui/layers/ingame/inventory/InventoryGrid.java [hunk 1]
```diff
@@ -52,14 +52,14 @@
...
-        int numSlots = InventoryUtils.getSlotCount(getTargetEntity()) - getCellOffset();
+        int numSlots = getNumSlots();
 
         // allow the UI to shrink the cell count if the inventory shrinks
         if (numSlots < cells.size()) {
-            cells.clear();
-        }
-
-        if (numSlots > cells.size() && cells.size() < getMaxCellCount()) {
+            for(int i = cells.size()-1; i >= 0; --i) {
```
Comment (immortius) [inline engine/src/main/java/org/terasology/rendering/nui/layers/ingame/inventory/InventoryGrid.java]: Can you run Checkstyle and address any issues it reports please? I can see a couple of minor style issues (mostly missing spaces).

**Comment by immortius** [comment]
This change might be rendered partially unnecessary with the fix to the number of inventory slots a player has (the inventory size should only be 40 cells with 10 on the toolbar).

**Comment by immortius** [comment]
That said, as long as the layout looks fine with the correct number of cells and adapts nicely to different resolutions, I'm happy with the changes.

**Comment by qreeves** [comment]
Apologies for the style issues, I tried to replicate the existing standard but obviously have some habits I missed. Should be fixed now.

The new layout should fit better on a variety of screens. I use 1280x720, so I had issues seeing the old one; the horizontal cell limit caused the number of vertical cells to exceed my screen dimensions, causing all sorts of terrible problems accessing items and overlapping other UI elements like the toolbar.

**Comment by Cervator** [comment]
Thanks for the tweaks! :-)

Unsure if this was just sort of a random tweak or if you're interested in GUI stuff in general. Seeing the inventory-chest display horizontal reminded me of a couple old threads talking about GUI placements and use in different cases

http://forum.terasology.org/threads/gui-where-to-put-what.623/
http://forum.terasology.org/threads/miniions.292/page-5#post-2849 (just a post with some mockups - main thing I was remembering: a split screen for trading with local player on bottom, remote player on top)

Both are pretty old and outdated at a technical level, but it is more about the design ideas. Also a bit related is http://forum.terasology.org/threads/mouse-button-behavior.704/

Just wanted to throw that in here in case of interest :D

Also a quick GitHub protip: Avoid merging _into_ topic branches, as you get excess merge commits like here, and in worse cases may end up reintroducing old commits as new, muddying up history - in this case it worked out fine though! Instead you can either leave the topic branch a little old as we'll do the merge to main develop, or try your hand at git rebase - although that can be tricky itself :-)

**Pull request merged**